signer/src/block_observer.rs

```diff
@@ -170,12 +171,27 @@ where
                     )
                     .increment(1);
 
-                    if let Err(error) = self.process_bitcoin_blocks_until(block_hash).await {
-                        tracing::warn!(%error, %block_hash, "could not process bitcoin blocks");
+                    let latest_processed = self.process_bitcoin_blocks_until(block_hash).await;
+
+                    match latest_processed {
+                        Ok(block_ref) => {
+                            if let Some(block_ref) = block_ref {
+                                if let Err(error) =
+                                    self.process_stacks_blocks(block_ref.block_height).await
+                                {
+                                    tracing::warn!(%error, "could not process stacks blocks");
+                                }
```

Comment:
We can now pass the chain tip down to `check_pending_dkg_shares` and `update_signer_state`, and remove the `get_bitcoin_block` calls in both of them.